plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts

```diff
@@ -153,6 +154,18 @@ export class OAuthAdapter implements AuthProviderRouteHandlers {
         }
       }
 
+      // Update cookie options to reflect any changes for the appOrigin
+      const updatedCookieOptions = this.options.cookieConfigurer({
+        providerId: this.options.providerId,
+        baseUrl: this.options.baseUrl,
+        callbackUrl: this.options.callbackUrl,
+        appOrigin,
+      });
+      this.baseCookieOptions = {
```

Comment:
Can't really rely on mutating the options here tbh, we'll run into issues with concurrent users from multiple different domains. I just realized we should be able to rely on the `Origin` header for the refresh and logout endpoints though! Afaik it'll be present in all the cases where we need it, and we'd otherwise fall back to the current behavior i.e. `lax`

Reply:
Ok, I think I got it. I'm now using the `Origin` header in `refresh` and `logout`. And the origin derived from `state` (if present) in the `frameHandler` which I believe make sense as it's used there for the `isOriginAllowed` check.
Let me know if I misunderstood something! 👍